plugins/amazonq/src/test/kotlin/PluginAmazonQJvmBinaryCompatabilityTest.kt

```diff
@@ -0,0 +1,72 @@
+        //      45: getfield      #40                 // Field software/aws/toolkits/jetbrains/services/codewhisperer/customization/CodeWhispererCustomization.arn:Ljava/lang/String;
+        //      58: getfield      #49                 // Field software/aws/toolkits/jetbrains/services/codewhisperer/customization/CodeWhispererCustomization.name:Ljava/lang/String;
+
+        //   public static final void setCustomization(com.intellij.openapi.project.Project);
+        //       6: getstatic     #72                 // Field migration/software/aws/toolkits/jetbrains/services/codewhisperer/customization/CodeWhispererModelConfigurator.Companion:Lmigration/software/aws/toolkits/jetbrains/services/codewhisperer/customization/CodeWhispererModelConfigurator$Companion;
+        //       9: invokevirtual #78                 // Method migration/software/aws/toolkits/jetbrains/services/codewhisperer/customization/CodeWhispererModelConfigurator$Companion.getInstance:()Lmigration/software/aws/toolkits/jetbrains/services/codewhisperer/customization/CodeWhispererModelConfigurator;
+        //      19: invokestatic  #82                 // InterfaceMethod migration/software/aws/toolkits/jetbrains/services/codewhisperer/customization/CodeWhispererModelConfigurator.listCustomizations$default:(Lmigration/software/aws/toolkits/jetbrains/services/codewhisperer/customization/CodeWhispererModelConfigurator;Lcom/intellij/openapi/project/Project;ZILjava/lang/Object;)Ljava/util/List;
+        //      140: invokeinterface #118,  3          // InterfaceMethod migration/software/aws/toolkits/jetbrains/services/codewhisperer/customization/CodeWhispererModelConfigurator.switchCustomization:(Lcom/intellij/openapi/project/Project;Lsoftware/aws/toolkits/jetbrains/services/codewhisperer/customization/CodeWhispererCustomization;)V
+
+        // CodeWhispererModelConfigurator.getInstance()
+        assertThat(Class.forName("migration.software.aws.toolkits.jetbrains.services.codewhisperer.customization.CodeWhispererModelConfigurator\$Companion").getMethod("getInstance").returnType)
+            .isEqualTo(Class.forName("migration.software.aws.toolkits.jetbrains.services.codewhisperer.customization.CodeWhispererModelConfigurator"))
+
+        val modelConfigurator = Class.forName("migration.software.aws.toolkits.jetbrains.services.codewhisperer.customization.CodeWhispererModelConfigurator")
+        // CodeWhispererModelConfigurator.listCustomizations(...)
+        // type erasure :/
+        assertThat(
+            modelConfigurator.getMethod(
+                "listCustomizations\$default",
+                modelConfigurator,
+                Class.forName("com.intellij.openapi.project.Project"),
+                // can't request primitive type using reflection
+                java.lang.Boolean.TYPE,
+                Integer.TYPE,
+                Class.forName("java.lang.Object")
```

Comment:
Where are these parameter types coming from? [The method signature](https://github.com/aws/aws-toolkit-jetbrains/blob/dd25f96bb575e24ed71e8d826718ac8a88d68723/plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/customization/CodeWhispererModelConfigurator.kt#L89) only shows `Project` and `Boolean`.